jhbuild/modtypes/__init__.py

```diff
@@ -300,6 +302,47 @@ def _process_install_files(self, installroot, curdir, prefix, errors):
                 errors.append(str(e))
         return num_copied
 
+    def _do_strip(self, destdir_prefix, contents):
+        # first find exec
+        files_to_strip = []
+
+        # HACK from orininal bash script.
+        excluded_packages = ['scipy', 'PIL']
+        excluded_libs = ['libQt5']
+        excluded_file_info = ['ELF', ', not stripped']
+
+        # filter out files to strip
+        for filename in contents:
+            if os.access(filename, os.X_OK) or 'so' in filename.split(os.path.extsep):
+                splited_filename = filename.split(os.path.sep)
+                if any([s == e for s in splited_filename for e in excluded_packages]):
+                    continue
+                if any([e in filename for e in excluded_libs]):
+                    continue
+                file_info = subprocess.check_output(['file', '-b', filename])
+                if any([e in file_info for e in excluded_file_info]):
```

Comment:
Is this correct? I think all the file will have `ELF` in the info, no?

Reply:
I rather you do the `if` conditions in the loop without the `excluded_*` variables. Because:

1. most of these hacks will need to be removed
2. it makes it clearer to me what condition you are testing for, some of the current conditions are wrong